warn-stranded.lua

```diff
@@ -58,73 +64,67 @@ local function addId(text, unit)
     return text..'|'..unit.id..'| '
 end
 
--- Uses persistent API. Low-level, deserializes 'warnStrandedIgnored' key and
---   will return an initialized empty warnStrandedIgnored table if needed.
--- Performance characterstics unknown of persistent API
-local function deserializeIgnoredUnits()
-    local currentIgnore = dfhack.persistent.get('warnStrandedIgnore')
-    if currentIgnore == nil then return {} end
+-- ===============================================
+--              Persistence API
+-- ===============================================
+-- Optional refresh parameter forces us to load from API instead of using cache
+
+-- Uses persistent API. Low-level, gets all entries currently in our persistent table
+--   will return an empty array if needed. Clears and adds entries to our cache.
+-- Returns the new global ignoresCache value
+local function getIgnoredUnits()
+    local ignores = dfhack.persistent.get_all(scriptPrefix)
+    if ignores == nil then return {} end
```

Comment:
this will return an empty set when there is no saved data to load, but will not set `ignoresCache`. Looking at the callers of this function, it looks like it is always expected that `ignoresCache` is set/updated after calling `getIgnoredUnit()`. This is especially important in the `onStateChange` hook, where a failure to clear the old data out of `ignoresCache` may cause strange behavior in the newly loaded fort.

on another note, it looks like the return value of `getIgnoredUnits()` is never used. should this instead be called `loadIgnoredUnits()`?

Reply:
Interestingly I did test and loading a fort with ignores followed by loading one without any does clear it! Good notes though, I'll get this updated tonight (heading to the ren fair!)